packages/@biomejs/biome/configuration_schema.json

```diff
@@ -2081,6 +2095,34 @@
 			},
 			"additionalProperties": false
 		},
+		"NoUndeclaredDependenciesConfiguration": {
+			"anyOf": [
+				{ "$ref": "#/definitions/RulePlainConfiguration" },
+				{ "$ref": "#/definitions/RuleWithNoUndeclaredDependenciesOptions" }
+			]
+		},
+		"NoUndeclaredDependenciesOptions": {
+			"description": "Rule's options",
+			"type": "object",
+			"properties": {
+				"devDependencies": {
+					"description": "If set to `false`, then the rule will show an error when `devDependencies` are imported. Defaults to `true`.",
+					"default": "Available",
```

Comment:
Can we use `true` as the defaule value here?

Reply:
You have two options: 
- mark the options with the `Option<>`. Then when extracting the options, you'll do `unwrap_or_default()` so you'll get the `true` default value (`Available`).
- if you don't want to use `Option<>`, you need to dumb down the types and the doing a custom deserialisation: 
	```rs
	enum Option {
		Bool(bool),
		Patterns(Vec<String>)
	}
	```

Reply:
The latter seems not working :cry:

<img width="466" alt="image" src="https://github.com/user-attachments/assets/6993844b-a10b-402c-abdc-55683aef2141">

Reply:
Have you added a custom serialisation function? Because that's what you need to do

Reply:
Added a custom `serde::Serialize` implementation and it works!: [fc2424a](https://github.com/biomejs/biome/pull/4376/commits/fc2424a6c7b7512357db1ea83a2f80b42d9e8daf)

However do we still need to implement both `serde::Deserialize` and `Deserializable`? Config options seems read by `Deserialize` so we need only `Deserializable` and `serde::Serialize`?

Reply:
I think you should choose the approach proposed by @ematipico.
You can use the struct attribute `serde(untagged)` to ensure that serde flatten the union. See the [serde docs](https://serde.rs/enum-representations.html#untagged).

Reply:
@ematipico @Conaclos Sorry for my misundersting. Changed to use `#[serde(untagged)]` in [5c3bdc0](https://github.com/biomejs/biome/pull/4376/commits/5c3bdc0385d4f0ce7eb9abb429bb1475d8535425).

Reply:
No problem. This is hard stuff :)